src/Backend/Core/Engine/Model.php

```diff
@@ -198,18 +198,24 @@ public static function deleteExtra($module = null, $type = null, array $data = n
 
         // loop found extras
         foreach ($extras as $extra) {
+            $delete = true;
+
             // match by parameters
             if ($data !== null && $extra['data'] !== null) {
                 $extraData = (array) unserialize($extra['data']);
 
-                // skip extra if parameters do not match
-                if (count(array_intersect($data, $extraData)) !== count($data)) {
-                    continue;
+                // delete extra if parameters do not match
```

Comment:
I'd change the comment here to "**do not** delete extra if the parameters do not match"